code/game/machinery/computer/message.dm

```diff
@@ -487,10 +542,71 @@
 					src.screen = 4
 
 //			to_chat(usr, href_list["select"])
-
+		if (href_list["switchboard"])
+			src.screen = 5
+		if (href_list["stopCall"])
+			var/obj/landline/L = locate(href_list["stopCall"])
+			if(L)
+				if(L.calling)
+					L.calling.last_call_log += text("call disconnected by operator<BR>")
+					L.calling.calling = null
+					if(L.calling.linked_phone)
+						playsound(L.calling.linked_phone, switchboard_sound2, 100, 1)
+				L.last_call_log += text("call disconnected by operator<BR>")
+				if(L.linked_phone)
+					playsound(L.linked_phone, switchboard_sound2, 100, 1)	
+				L.calling = null
+			updateUsrDialog()
+			playsound(src, switchboard_sound2, 100, 1)
+		if (href_list["manualCall"])
+			var/obj/landline/L = locate(href_list["manualCall"])
+			if(L && linking && (L != linking))
+				if(L.calling || linking.calling)
+					return
+				L.calling = linking
+				linking.calling = L
+				L.ring_loop()
+				linking.ring_loop()
+				var/obj/machinery/requests_console/L_console = L.attached_to
+				var/obj/machinery/requests_console/linking_console = linking.attached_to
+				L.last_call_log += text("connected to [linking_console.department] by operator<BR>")
+				linking.last_call_log += text("connected to [L_console.department] by operator<BR>")
+				if(L.calling.linked_phone)
+					playsound(L.calling.linked_phone, switchboard_sound, 100, 1)
+				if(L.linked_phone)
+					playsound(L.linked_phone, switchboard_sound, 100, 1)
+				playsound(src, switchboard_sound, 100, 1)
+				linking = FALSE
+		if (href_list["setLink"])
+			var/obj/landline/L = locate(href_list["setLink"])
+			if(!L || linking)
+				return
+			linking = L
+			playsound(src, switchboard_sound, 100, 1)
+			updateUsrDialog()
 		if (href_list["back"])
 			src.screen = 0
-
+		if (href_list["clearCurrentLine"])
+			if(switchboard_headset && switchboard_headset.linked_landline)
```

Comment:
can you at least turn `switchboard_headset.linked_landline` into a var to spare any future coder's eyes